src/core/index.js

```diff
@@ -108,6 +108,8 @@ class IPFS extends EventEmitter {
     this._preload = preload(this)
     this._mfsPreload = mfsPreload(this)
     this._ipns = new IPNS(null, this)
+    this._print = options.silent ? (() => {}) : ((msg) => console.log(msg))
```

Comment:
I think it might be a good idea to move printed messages into the debug log if `silent=true`, so if you specify `DEBUG=jsipfs* jsipfs daemon --silent` you'll still be able to access that information - what do you think?

```suggestion
    this._print = this._options.silent ? this.log : console.log
```

Also, we should pass the `console.log` funciton directly here so that we can log multiple args.

Reply:
When I pass the `--silent` as an option in the command line I'm still receiving Swarm addresses as output in the terminal.

Reply:
Try logging out the value of `this._options.silent` here and `argv.silent` in `src/cli/utils.js `. Maybe it's not being passed in correctly?

Reply:
Did you figure this out in the end?

Reply:
Haven't had a chance to work on this in the past couple of days, but plan on getting back to this this afternoon.